Refresh documentation (and some tests) of the Hibernate Search + ORM extension

yrodiere opened the pull request:
I finally got around to having a long hard look at this while I was working on #26182 and had to essentially copy and adapt the documentation.

This reorganizes a few sections, adds some documentation, and updates a few examples.

We will need to update quickstarts to match; I'll add a link to that PR here soon. EDIT: here it is: https://github.com/quarkusio/quarkus-quickstarts/pull/1394

yrodiere committed:
Move Hibernate Search DevUI tests to hibernate-search-orm-elasticsearch/deployment

Because we'll need to test the Standalone extension too, and we can't
keep adding dependencies to a single integration test module.

yrodiere committed:
Refresh the documentation of hibernate-search-orm-elasticsearch

Reorganize sections, update wording to make it clear it's about the
integration of Hibernate Search with Hibernate ORM.

yrodiere committed:
Improve documentation of pluggable components for Hibernate Search

gsmet reviewed (approved):
LGTM. I added a few minor comments, feel free to ignore if you disagree.

yrodiere commented:
Thanks @gsmet , I addressed your comments.

Ready to merge this and https://github.com/quarkusio/quarkus-quickstarts/pull/1394 as soon as the build passes.
